test/math/SafeMath.test.js

```diff
@@ -5,142 +5,396 @@ const { expect } = require('chai');
 
 const SafeMathMock = artifacts.require('SafeMathMock');
 
+async function deepExpect (value, expected) {
```

Comment:
There is technically a problem with this function. It iterates using for..in, but recurses if the value has `Symbol.iterator`. The thing is `Symbol.iterator` is used for for..of loops, whereas for..in iterates over `Object.keys`.

IMO we should have a simpler function to compare the kinds of objects returned by the try functions. `expectEqualWithFlag` and no recursion. Plus `testCommutativeIterable` -> `testCommutativeWithFlag`.